common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/PurchasedProductsFetcher.kt

```diff
@@ -21,7 +23,15 @@ class PurchasedProductsFetcher(
         onSuccess: (List<PurchasedProduct>) -> Unit,
         onError: (PurchasesError) -> Unit
     ) {
-        val productEntitlementMapping = deviceCache.getProductEntitlementMapping()
+        val productEntitlementMapping = deviceCache.getProductEntitlementMapping() ?: run {
```

Comment:
Note that we will still calculate the offline customer info if it's empty, we just want to fail if there is no stored mapping, since we really can't perform an accurate calculation